c10/util/BFloat16-math.h

```diff
@@ -19,193 +18,201 @@ struct is_reduced_floating_point
 template <typename T>
 constexpr bool is_reduced_floating_point_v =
     is_reduced_floating_point<T>::value;
+} // namespace c10
+
+namespace std {
+
+#if !defined(FBCODE_CAFFE2) && !defined(C10_NODEPRECATED)
+using c10::is_reduced_floating_point;
+using c10::is_reduced_floating_point_v;
+#endif // !defined(FBCODE_CAFFE2) && !defined(C10_NODEPRECATED)
```

Comment:
To be frank, I don't see much use of that template outside of PyTorch, per https://github.com/search?type=code&q=is_reduced_floating_point so IMO it's OK to remove it (probably as followup PR)
But is it possible to slap deprecated attribute to this using pattern?

Reply:
> is it possible to slap deprecated attribute to this using pattern

it looks like the existing `!defined(FBCODE_CAFFE2) && !defined(C10_NODEPRECATED)` don't tag the items as `[[deprecated]]` (CC @r-barnes), so I'll try adding that in a follow-up.

Reply:
FWIW, `[[deprecated]]` is allowed on `typedef`s (and the `using` syntax for them), but not on plain old `using` declarations like these.

Reply:
oh, and every single `!defined(FBCODE_CAFFE2) && !defined(C10_NODEPRECATED)` item is something we can't `[[deprecated]]`, so there's nothing to follow up on.

Reply:
Yeah, @swolchok, I added `FBCODE_CAFFE2` so we could avoid internal usages of things we'd deprecated and `C10_NODEPRECATED` so that external users could check if they were using things that can't be marked as deprecated.